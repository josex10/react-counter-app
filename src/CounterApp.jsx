import { useState } from 'react';
import PropTypes from 'prop-types';
export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);
  const handleAdd = () => setCounter(counter + 1);
  const handleSubtract = () => setCounter(counter - 1);
  const handleReset = () => setCounter(value);
  return (
    <>
      <h1 aria-label="heading-title">Counter App</h1>
      <h2 aria-label="heading-counter">{counter}</h2>

      <button aria-label="btn-sum" onClick={handleAdd}>
        +1
      </button>
      <button aria-label="btn-sub" onClick={handleSubtract}>
        -1
      </button>
      <button aria-label="btn-reset" onClick={handleReset}>
        Reset
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number
};

CounterApp.defaultProps = {
  value: 0
};
