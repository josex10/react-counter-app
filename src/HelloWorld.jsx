import PropTypes from 'prop-types';

export const HelloWorld = ({ title, subTitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

HelloWorld.propTypes = {
  subTitle: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};

HelloWorld.defaultProps = {
  name: 'Jose Manuel Badilla Porras - default',
  subTitle: 'Without subtitle prop - default'
};
