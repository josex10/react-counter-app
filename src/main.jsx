import React from 'react';
import ReactDOM from 'react-dom/client';
import { HelloWorld } from './HelloWorld';
import { CounterApp } from './CounterApp';
import './style.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <HelloWorld />
  </React.StrictMode>
);
