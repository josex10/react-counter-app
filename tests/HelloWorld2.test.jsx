import { render, screen } from '@testing-library/react';
import { HelloWorld } from '../src/HelloWorld';

describe('Testing component Hello World 2', () => {
  const title = 'Hello World from test';
  const subtitle = 'Hello World from test - subtitle';

  test('Should be a match with the snapshot', () => {
    const { container } = render(<HelloWorld title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('Should be a meesage "Hello World from test"', () => {
    render(<HelloWorld title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test('Should be the title on h1 tag', () => {
    render(<HelloWorld title={title} />);
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test('should be show the subtiitle sent by props', () => {
    render(<HelloWorld title={title} subTitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(1);
  });
});
