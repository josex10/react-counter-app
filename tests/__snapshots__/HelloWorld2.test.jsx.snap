// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component Hello World 2 Should be a match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hello World from test
  </h1>
  <p>
    Without subtitle prop - default
  </p>
  <p>
    Jose Manuel Badilla Porras - default
  </p>
</div>
`;
